### PR TITLE
Reconcile Loadbalancers when we update instance group

### LINKED ISSUE #154: error running "make create cluster"
/kind bug

root@capd-2:~/cluster-api-provider-gcp# make create-cluster
kind create cluster --name=clusterapi
Creating cluster "clusterapi" ...
 ✓ Ensuring node image (kindest/node:v1.15.3) 🖼
 ✓ Preparing nodes 📦
 ✓ Creating kubeadm config 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Cluster creation complete. You can now use the cluster with:

export KUBECONFIG="$(kind get kubeconfig-path --name="clusterapi")"
kubectl cluster-info
# Apply provider-components.
root@capd-2:~/cluster-api-provider-gcp# make create-cluster
kind create cluster --name=clusterapi
Creating cluster "clusterapi" ...
 ✓ Ensuring node image (kindest/node:v1.15.3) 🖼
 ✓ Preparing nodes 📦
 ✓ Creating kubeadm config 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Cluster creation complete. You can now use the cluster with:

export KUBECONFIG="$(kind get kubeconfig-path --name="clusterapi")"
kubectl cluster-info
# Apply provider-components.
kubectl \
        --kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
        create -f examples/_out/provider-components.yaml
namespace/cabpk-system created
namespace/capg-system created
namespace/capi-system created
customresourcedefinition.apiextensions.k8s.io/clusters.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/gcpclusters.infrastructure.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/gcpmachines.infrastructure.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/gcpmachinetemplates.infrastructure.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/kubeadmconfigs.bootstrap.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/kubeadmconfigtemplates.bootstrap.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/machinedeployments.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/machines.cluster.x-k8s.io created
customresourcedefinition.apiextensions.k8s.io/machinesets.cluster.x-k8s.io created
role.rbac.authorization.k8s.io/cabpk-leader-election-role created
role.rbac.authorization.k8s.io/capg-leader-election-role created
role.rbac.authorization.k8s.io/capi-leader-election-role created
clusterrole.rbac.authorization.k8s.io/cabpk-manager-role created
clusterrole.rbac.authorization.k8s.io/cabpk-proxy-role created
clusterrole.rbac.authorization.k8s.io/capg-manager-role created
clusterrole.rbac.authorization.k8s.io/capg-proxy-role created
clusterrole.rbac.authorization.k8s.io/capi-manager-role created
rolebinding.rbac.authorization.k8s.io/cabpk-leader-election-rolebinding created
rolebinding.rbac.authorization.k8s.io/capg-leader-election-rolebinding created
rolebinding.rbac.authorization.k8s.io/capi-leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/cabpk-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/cabpk-proxy-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/capg-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/capg-proxy-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/capi-manager-rolebinding created
secret/capg-manager-bootstrap-credentials created
service/cabpk-controller-manager-metrics-service created
service/capg-controller-manager-metrics-service created
deployment.apps/cabpk-controller-manager created
deployment.apps/capg-controller-manager created
deployment.apps/capi-controller-manager created
# Create Cluster.
kubectl \
        --kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
        create -f examples/_out/cluster.yaml
cluster.cluster.x-k8s.io/test1 created
gcpcluster.infrastructure.cluster.x-k8s.io/test1 created
# Create control plane machine.
kubectl \
        --kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
        create -f examples/_out/controlplane.yaml
kubeadmconfig.bootstrap.cluster.x-k8s.io/test1-controlplane-0 created
kubeadmconfig.bootstrap.cluster.x-k8s.io/test1-controlplane-1 created
kubeadmconfig.bootstrap.cluster.x-k8s.io/test1-controlplane-2 created
machine.cluster.x-k8s.io/test1-controlplane-0 created
machine.cluster.x-k8s.io/test1-controlplane-1 created
machine.cluster.x-k8s.io/test1-controlplane-2 created
gcpmachine.infrastructure.cluster.x-k8s.io/test1-controlplane-0 created
gcpmachine.infrastructure.cluster.x-k8s.io/test1-controlplane-1 created
gcpmachine.infrastructure.cluster.x-k8s.io/test1-controlplane-2 created
# Get KubeConfig using clusterctl.
bin/clusterctl \
        alpha phases get-kubeconfig -v=3 \
        --kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
        --namespace=default \
        --cluster-name=test1
I0912 01:04:20.704681     789 getkubeconfig.go:40] Getting target cluster kubeconfig.
I0912 01:04:20.704760     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:04:30.708354     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:04:40.708401     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:04:50.708360     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:05:00.708373     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:05:10.708350     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:05:20.708428     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:05:30.708362     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:05:40.708438     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:05:50.708385     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0912 01:06:00.708375     789 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
# Apply addons on the target cluster, waiting for the control-plane to become available.
bin/clusterctl \
        alpha phases apply-addons -v=3 \
        --kubeconfig=./kubeconfig \
        -a examples/addons.yaml
I0912 01:06:00.787402    3368 clusterclient.go:237] Waiting to acquire client...
F0912 01:06:00.787708    3368 alpha_phase_apply_addons.go:50] unable to create cluster client: failed to acquire new client: Get https://34.102.233.46:443/api?timeout=32s: EOF
Makefile:262: recipe for target 'create-cluster' failed
make: *** [create-cluster] Error 1

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by dims**
Change-Id: Id06a4f2a44d6aa5cd9534bf63ba2c200d27c45ac

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #154

### CODE REVIEW

**Comment by dims** [comment]
Looking at activity, we create the first vm at `1:47 PM`, but the backend gets updated only at `1:56`, so until then even if the API server is up there is no traffic being sent to it. So let's call reconcile as soon as we add instances to instance group

Activity_–_Home_–_kubernetes_developm…_–_Google_Cloud_Platform-001

**Comment by dims** [comment]
way faster than before :)

```
bin/clusterctl \
	alpha phases get-kubeconfig -v=3 \
	--kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
	--namespace=default \
	--cluster-name=test1
I0918 19:35:25.862148   19601 getkubeconfig.go:40] Getting target cluster kubeconfig.
I0918 19:35:25.862273   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:35:36.534750   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:35:46.534795   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:35:56.534413   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:36:06.534819   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:36:16.534762   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:36:26.534822   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:36:36.534843   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:36:46.534469   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
I0918 19:36:56.534828   19601 getkubeconfig.go:65] Waiting for kubeconfig from Secret "test1-kubeconfig" in namespace "default" to become available...
# Apply addons on the target cluster, waiting for the control-plane to become available.
bin/clusterctl \
	alpha phases apply-addons -v=3 \
	--kubeconfig=./kubeconfig \
	-a examples/addons.yaml
I0918 19:36:56.612788   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:37:06.661787   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:37:16.663603   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:37:26.665392   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:37:36.662341   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:37:46.664818   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:37:56.665732   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:38:06.666655   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:38:16.659657   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:38:26.664201   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:38:36.661385   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:38:46.663439   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:38:56.667600   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:39:06.673548   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:39:16.660291   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:39:26.661528   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:39:36.663138   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:39:46.681034   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:39:56.663482   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:40:06.671545   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:40:16.663578   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:40:26.673317   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:40:36.661879   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:40:46.665102   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:40:56.661771   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:41:06.659841   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:41:16.664629   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:41:26.666385   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:41:36.665210   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:41:46.663895   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:41:56.664723   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:42:06.670912   21837 clusterclient.go:239] Waiting to acquire client... server not yet available: Get https://34.102.174.163:443/api?timeout=32s: EOF
I0918 19:42:16.819603   21837 applyaddons.go:25] Applying Addons
I0918 19:42:16.819636   21837 clusterclient.go:786] Waiting for kubectl apply...
# Create a worker node with MachineDeployment.
kubectl \
	--kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
	create -f examples/_out/machinedeployment.yaml
kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/test1-md-0 created
machinedeployment.cluster.x-k8s.io/test1-md-0 created
gcpmachinetemplate.infrastructure.cluster.x-k8s.io/test1-md-0 created
```

**Comment by dims** [comment]
evidence of the backend service getting added as soon as vm is created.

Activity_–_Home_–_kubernetes_developm…_–_Google_Cloud_Platform-004

**Comment by vincepri** [comment]
I wonder if instead we should make GCPCluster watch changes on GCPMachines?

**Comment by vincepri** [comment]
Actually, it does seem we are watching GCPMachines using `Owns(&infrav1.GCPMachine{}).` I wonder why calling it twice changes it?

File: controllers/gcpmachine_controller.go [hunk 1]
```diff
@@ -365,6 +365,12 @@ func (r *GCPMachineReconciler) reconcileLBAttachment(machineScope *scope.Machine
 		return errors.Wrapf(err, "failed to add instance to group")
 	}
 
+	computeSvc := compute.NewService(clusterScope)
```
Comment (vincepri) [inline controllers/gcpmachine_controller.go]: Can we add a TODO to remove this and debug why we'd need to reconcile it here?

Comment (dims) [reply controllers/gcpmachine_controller.go]: will do!

Comment (vincepri) [reply controllers/gcpmachine_controller.go]: Or an issue, let's merge this for now and debug later :)

Comment (dims) [reply controllers/gcpmachine_controller.go]: cool opening

**Pull request merged**